lib/player.js

```diff
@@ -1868,8 +1868,22 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // Pick the initial streams to play.
     const initialVariant = this.chooseVariant_();
     goog.asserts.assert(initialVariant, 'Must choose an initial variant!');
-    this.switchVariant_(initialVariant, /* fromAdaptation= */ true,
-        /* clearBuffer= */ false, /* safeMargin= */ 0);
+
+    // However, we would skip switch to initial variant
+    // if user already pick variant track (via selectVariantTrack api)
+    let activeVariantTrack = null;
+
+    for (const variantTrack of this.getVariantTracks()) {
+      if (variantTrack.active) {
+        activeVariantTrack = variantTrack;
+        break;
+      }
+    }
+
+    if (! activeVariantTrack) {
```

Comment:
Looks like this doesn't match the style.  Please remove the space between `!` and `activeVariantTrack`, and please also run `python build/check.py` to run the linter yourself.